### PR TITLE
Draw sprites overlapping into visible sectors

### LINKED ISSUE #1537: R_DrawSprite: Enhancement to stop Things from disappearing due to "anchor" sector not being in line of sight
There is an issue across source ports where sprites will abruptly disappear if the "center" point of a Thing is located on a sector that isn't currently visible to the player, even if the vast majority of the Thing would be visible to the player. This behavior is not unique to Woof, but improving the function here will likely enable it to be improved across all related sourceports.

I have attached a demonstration WAD here. There are three mancubi facing away from the player, each with a small column between them and the player. From left to right, I will call these Mancubus 1, Mancubus 2, and Mancubus 3. Mancubus 1 is in the same sector as the player; Mancubus 2 is in a small child sector; and Mancubus 3 is in an even smaller child sector. By strafing left and right and approaching the columns, we can observe different behaviors:

1. Mancubus 1 will always be shown without issue, regardless of the player's position behind the pillars.
2. Mancubus 2 will generally always be visible when the player views them from a distance, but if the player moves close enough to the pillar behind it, the Mancubus will abruptly disappear.
3. Mancubus 3 will very easily pop in and pop out as the player strafes left and right, even when the player is not close to the pillar behind it.

While I'm not sure what form an improvement would take from a technical standpoint, in general, it seems like we could try to add some additional checks before determining that a sprite doesn't need to be drawn: in addition to checking the center position of the Thing, perhaps we should also check for visibility based on one or more of the Thing's four corners if the center is not visible. While this would not catch all cases, it would certainly result in much less pop-in and pop-out, and thus be a net improvement.

[DrawTest.zip](https://github.com/fabiangreffrath/woof/files/14412889/DrawTest.zip)

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #1537

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
About 30 visible sprites more and 1 fps less in the opening scene of comatose.wad MAP01.

**Comment by fabiangreffrath** [comment]
NB, this needs to be made optional, map creators may use this trick to deliberately hide monsters from the player's sight.

**Comment by fabiangreffrath** [comment]
How would we call this in the menu? "Improve Sprite Visibility"?

**Comment by MrAlaux** [comment]
> How would we call this in the menu?

"Prevent Sprite Flickering"?

**Comment by rfomin** [comment]
> NB, this needs to be made optional, map creators may use this trick to deliberately hide monsters from the player's sight.

> How would we call this in the menu? "Improve Sprite Visibility"?

Honestly, I don't like options like this in the menu, no one would ever understand what it means. I think we should add it as a config/OPTIONS.lmp item, perhaps with a `comp_` name.

**Comment by ceski-1** [comment]
> Honestly, I don't like options like this in the menu, no one would ever understand what it means. I think we should add it as a config/OPTIONS.lmp item, perhaps with a `comp_` name.

I agree that this probably doesn't belong in the menu, but if it was added somewhere I guess it's a gray area between "compatibility-breaking" and cosmetic (general > display?).

**Comment by fabiangreffrath** [comment]
I'd put it right next to "Improve hit detection", but I still cannot imagine a name for it that's short enough to fit in a single line but not confusing at the same time.

**Comment by rfomin** [comment]
My suggestion would be to make it config only and enable it by default, and see if anyone complains. It's only visual, and I don't know of any PWADs that use it intentionally. Also, this is fixed in GZDoom, there are very few PWADs that are intentionally incompatible with GZDoom.

**Comment by fabiangreffrath** [comment]
Still not sure what to do about this...

**Comment by ceski-1** [comment]
I agree with rfomin's suggestion. Config only, enabled by default. Change if needed based on user feedback. I can't think of a good name for it.

File: src/r_things.c [hunk 1]
```diff
@@ -737,6 +744,46 @@ void R_AddSprites(sector_t* sec, int lightlevel)
 
   for (thing = sec->thinglist; thing; thing = thing->snext)
     R_ProjectSprite(thing);
+
+  if (CRITICAL(draw_nearby_sprites))
```
Comment (rfomin) [inline src/r_things.c:748]: It doesn't affect compatibility, does it? Maybe `STRICTMODE` is enough.

**Review by rfomin — APPROVED** [review]
Thank you!

**Pull request merged**